SmartDeviceLink/SDLMutableDataQueue.h

```diff
@@ -0,0 +1,38 @@
+//
+//  SDLMutableDataQueue.h
+//  
+//
+//  Created by David Switzer on 3/2/17.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+@interface SDLMutableDataQueue : NSObject
```

Comment:
It specifically handles NSMutableData because it feeds an NSStream with an unknown amount of space available for writing. The data at the head of the queue needs to be consumed as space is available without changing its position in the queue.